src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -26,7 +32,34 @@ public DeleteCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_EMPTY_USAGE));
         }
 
-        return StringUtil.isName(args) ? parseByName(args) : parseByIndex(args);
+        if (StringUtil.isName(args)) {
+            return parseByName(args);
+        } else if ((StringUtil.isTag(args))) {
+            return parseByTag(args);
+        } else {
+            return parseByIndex(args);
+        }
+    }
+    private DeleteCommand parseByTag(String args) throws ParseException {
+        try {
+            String trimmedArgs = args.trim();
+            if (trimmedArgs.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_TAG_USAGE));
+            }
+            ArgumentMultimap argMultimap =
+                    ArgumentTokenizer.tokenize(args, PREFIX_TAG);
+            String[] tags = trimmedArgs.split(PREFIX_TAG.toString());
+            if (tags.length < 2) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_TAG_USAGE));
+            }
+            String[] tagKeywords = tags[1].trim().split("\\s+");
```

Comment:
I think better to put tags[1] in some variable and give a name to it so easier for the others to know what does it do.